Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt

```diff
@@ -41,23 +46,54 @@ object MainDestinations {
 }
 
 @Composable
-fun NavGraph(startDestination: String = MainDestinations.ONBOARDING_ROUTE) {
-    val navController = rememberNavController()
+fun NavGraph(
+    modifier: Modifier = Modifier,
+    finishActivity: () -> Unit = {},
+    navController: NavHostController = rememberNavController(),
+    startDestination: String = MainDestinations.COURSES_ROUTE,
+    showOnboardingInitially: Boolean = true
+) {
+    // Onboarding could be read from shared preferences.
+    val onboardingComplete = remember(showOnboardingInitially) {
```

Comment:
This code messed with my head. It's correct, but it took me a minute to work it out.

Reply:
Yeah... we need a better API for this